persistence-kafka/src/main/scala/com/evolutiongaming/kafka/flow/kafkapersistence/KafkaPersistenceModule.scala

```diff
@@ -138,22 +136,23 @@ object KafkaPersistenceModule {
       cache: Cache[F, String, ByteVector],
       producer: Producer[F]
     ): F[SnapshotPersistenceOf[F, KafkaKey, S, ConsRecord]] = {
-      LogOf[F].apply(classOf[SnapshotPersistenceOf[F, KafkaKey, S, ConsRecord]]).map { implicit log =>
-        implicit val producer_ = producer
-        val read               = KafkaSnapshotReadDatabase.of[F, S](snapshotTopicPartition.topic, key => cache.remove(key).flatten)
+      LogOf[F].apply(classOf[KafkaPersistenceModule[F, S]]).map { log =>
+        val read =
+          KafkaSnapshotReadDatabase.of[F, S](snapshotTopicPartition.topic, getState = key => cache.remove(key).flatten)
 
-        val snapshotsOf = new SnapshotsOf[F, KafkaKey, S] {
-          override def apply(key: KafkaKey): F[Snapshots[F, S]] =
-            for {
-              buffer <- Ref.of(none[Snapshot[S]]).map(_.stateInstance)
-            } yield Snapshots(
-              key = key,
-              database = SnapshotDatabase(
-                read  = read,
-                write = KafkaSnapshotWriteDatabase.of[F, S](snapshotTopicPartition)
-              ) withMetricsK metrics.snapshotDatabaseMetrics,
-              buffer = buffer
-            )
+        // A manual overriding of SnapshotsOf is required to pass a custom prefixed Log.
+        // Since both `SnapshotsOf.backedBy` and `Snapshots.of` are parameterized by a generic K (`KafkaKey` here),
+        // they would log the whole KafkaKey if Log.prefixed was to be called from inside.
```

Comment:
would it be a problem if we just used the kafka-key as log prefix? It would work with Cassandra persistence as well , right?

Reply:
The only option is using `.toString` which would result in something like this: `KafkaKey(my-application-id,my-group-id,my-topic-0,kafkaKey)`. It's not the worst, but it's some unnecessary information for a prefix, especially if `applicationId` and/or `groupId` are long. Do you think it makes sense to log all of it?

Reply:
thanks for sharing details, I don't think .toString is a good option, but I have another idea, let's discuss it

Reply:
Implemented the proposed changes, now log message looks like this in integration tests:
```
c.e.k.f.k.KafkaPersistenceModule - kafka-persistence-test-1 key1 deleted snapshot
```